phpBB/phpbb/controller/helper.php

```diff
@@ -212,12 +212,29 @@ public function error($message, $code = 500)
 	public function message($message, array $parameters = array(), $title = 'INFORMATION', $code = 200)
 	{
 		array_unshift($parameters, $message);
+		$message_text = call_user_func_array(array($this->user, 'lang'), $parameters);
+		$message_title = $this->user->lang($title);
+
+		if ($this->request->is_ajax())
+		{
+			global $refresh_data;
```

Comment:
Fine by me, even with the global.

At least the dirty code is restricted to one specific place.